### DISCUSSION TITLE
What will be most efficient?

### DISCUSSION THREAD

**Discussion by TheNodder** [xerial/sqlite-jdbc#701] [Q&A] [answered]
Hi,
Seems as the . last () is not implemented in the driver as SQLite does not support(?) this action; what would be fastest? Iterate .next() until the last record has been reached or an SQL-SELECT with parameters? Given is that no index are defined for the table.

Cheers,
Niclas

**Comment by gotson** [comment, marked as answer]
See https://github.com/xerial/sqlite-jdbc/issues/379#issuecomment-545138607

Sqlite itself is forward only, hence the lack of support for 'last()`.

As to what would be the most efficient, if depends on your query and data. In a small result set, maybe iterating would work. But in a big dataset it's pointless. If you can write a query to get the last value, that would be more efficient.

**Reply by TheNodder** [reply]
Thanks for your reply. I did what you suggested; a query to find the last() value. This works fine, the delay between the data to show and the end-users possibility to interact are more or less seamless
